administrator/modules/mod_sampledata/tmpl/default.php

```diff
@@ -14,41 +14,43 @@
 use Joomla\CMS\Uri\Uri;
 use Joomla\CMS\HTML\HTMLHelper;
 
-HTMLHelper::_('jquery.framework');
-HTMLHelper::_('script', 'mod_sampledata/sampledata-process.js', false, true);
+HTMLHelper::_('script', 'mod_sampledata/sampledata-process.js', ['version' => 'auto', 'relative' => true]);
 
 Text::script('MOD_SAMPLEDATA_CONFIRM_START');
 Text::script('MOD_SAMPLEDATA_ITEM_ALREADY_PROCESSED');
 Text::script('MOD_SAMPLEDATA_INVALID_RESPONSE');
 
-Factory::getDocument()->addScriptDeclaration('
-	var modSampledataUrl = "index.php?option=com_ajax&format=json&group=sampledata",
-		modSampledataIconProgress = "' . Uri::root(true) . '/media/system/images/ajax-loader.gif";
-');
+Factory::getDocument()->addScriptOptions(
+	'sample-data',
+	[
+		'url' => 'index.php?option=com_ajax&format=json&group=sampledata',
+		'icon' => Uri::root(true) . '/media/system/images/ajax-loader.gif'
```

Comment:
Since we have only 2 vars here I propose to use `data-url` and `data-icon`. The good part is that we don't need then the `Joomla.options` in the script, of course we still need the Joomla.request but that's another story.

Reply:
Why do we need those options at all? Seem pretty static to me.

Reply:
I can move the URL option to the JS, but I'm going to leave the icon option there till this has been merged: https://github.com/joomla/joomla-cms/pull/19677